app/javascript/components/lists/ListGroup.js

```diff
@@ -0,0 +1,47 @@
+import React, { useState, useEffect } from "react"
+import ListGroupButtons from "./ListGroupButtons"
+
+const ListGroup = (props) => {
+  const [lists, setLists] = useState(props.lists);
+
+  // var lists = props.lists;
+  const csrf = props.csrf;
+
+  const handleStateChange = ( listId ) => {
+    var newLists = [...lists];
+
+    newLists.map((list, index) =>{
+      let id = list.id;
+      if( id === listId ){
+        newLists.splice(index, 1);
+        setLists(newLists);
+        console.log("List: \"" + list.name + "\" removed");
+      }
+    })
+  }
+
+  const displayList = (list, index) => {
+    let activeClass = "";
+    if(index == 0){
+      activeClass = "active";
+    }
+    return(
+      <a className={"list-group-item list-group-item-action d-flex " + activeClass} id={"list-" + list.id + "-list"} data-bs-toggle="list" href={"#list-" + list.id} role="tab" aria-controls={"list-" + list.id}>
```

Comment:
Oh, and to fix the warning about not setting a key.
```suggestion
      <a key={list.id} className={"list-group-item list-group-item-action d-flex " + activeClass} id={"list-" + list.id + "-list"} data-bs-toggle="list" href={"#list-" + list.id} role="tab" aria-controls={"list-" + list.id}>
```